doc/source/cluster/key-concepts.rst

```diff
@@ -0,0 +1,50 @@
+.. include:: we_are_hiring.rst
+
+.. _cluster-key-concepts:
+
+Key Concepts
+============
+
+Cluster
+-------
+
+A Ray cluster is a set of one or more nodes that are running Ray and share the same :ref:`head node<cluster-node-types>`.
+
+.. _cluster-node-types:
+
+Node types
+----------
+
+A Ray cluster consists of a :ref:`head node<cluster-head-node>` and a set of :ref:`worker nodes<cluster-worker-node>`.
+
+.. image:: ray-cluster.jpg
+    :align: center
+    :width: 600px
+
+.. _cluster-head-node:
+
+Head node
+~~~~~~~~~
+
+The head node is the first node started by the :ref:`Ray cluster launcher<cluster-launcher>` when trying to launch a Ray cluster. Among other things, the head node hold the :ref:`Global Control Store (GCS)<memory>` and runs the :ref:`autoscaler<cluster-autoscaler>`. Once the head node is started, it will be responsible for launching any additional :ref:`worker nodes<cluster-worker-node>`. The head node itself will also execute tasks and actors to utilize its capacity.
```

Comment:
Nit question: Is the style to capitalize "cluster launcher" or not? Same question for "autoscaler."

Reply:
Personally I prefer not capitalizing because it's not that important of a product/concept across all of Ray but also not super adamant about it